internal/services/loganalytics/log_analytics_linked_storage_account_resource.go

```diff
@@ -131,6 +137,7 @@ func resourceLogAnalyticsLinkedStorageAccountRead(d *pluginsdk.ResourceData, met
 
 	d.Set("resource_group_name", id.ResourceGroupName)
 	d.Set("workspace_resource_id", linkedstorageaccounts.NewWorkspaceID(id.SubscriptionId, id.ResourceGroupName, id.WorkspaceName).ID())
+	d.Set("data_source_type", string(id.DataSourceType))
```

Comment:
since this is part of the id, we might want to do a state migration so we can remove the parse insensitively above?